cpp/log/signer.cc

```diff
@@ -59,7 +63,12 @@ std::string Signer::RawSign(const std::string& data) const {
   unsigned int sig_size = EVP_PKEY_size(pkey_.get());
   unsigned char* sig = new unsigned char[sig_size];
 
-  CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
+  if (synchronize_signing_) {
+    std::lock_guard<std::mutex> lock(lock_);
+    CHECK_EQ(1, EVP_SignFinal(&ctx, sig, &sig_size, pkey_.get()));
```

Comment:
Is this because some OpenSSL engines are not thread-safe? Is it sufficient to only protect the call to `EVP_SignFinal`?